dev/devicelab/bin/tasks/flutter_attach_test.dart

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+import 'package:flutter_devicelab/framework/adb.dart';
+import 'package:flutter_devicelab/framework/framework.dart';
+import 'package:flutter_devicelab/framework/utils.dart';
+
+Future<void> testReload(Process process, { Future<void> Function() onListening }) async {
+  print('Testing hot reload, restart and quit');
+  final Completer<Null> listening = new Completer<Null>();
+  final Completer<Null> ready = new Completer<Null>();
+  final Completer<Null> reloaded = new Completer<Null>();
+  final Completer<Null> restarted = new Completer<Null>();
+  final List<String> stdout = <String>[];
+  final List<String> stderr = <String>[];
+
+  if (onListening == null)
+    listening.complete();
+
+  int exitCode;
+  process.stdout
+      .transform(utf8.decoder)
+      .transform(const LineSplitter())
+      .listen((String line) {
+    print('attach:stdout: $line');
+    stdout.add(line);
+    if (line.contains('Listening') && onListening != null) {
+      listening.complete(onListening());
+    }
+    if (line.contains('To quit, press "q".'))
+      ready.complete();
+    if (line.contains('Reloaded '))
+      reloaded.complete();
+    if (line.contains('Restarted app in '))
```

Comment:
Duplication of strings here and lines 76-81

Reply:
done